pyproject.toml

```diff
@@ -63,7 +63,7 @@ classifiers = [
 ]
 requires-python = '>=3.8'
 dependencies = [
-    'typing-extensions>=4.6.1',
+    'typing-extensions>=4.9.0',
```

Comment:
One issue that I think is fairly serious with upgrading the minimum dependency version is that, as far as I can tell, tensorflow currently has a hard-pin on `typing_extensions==4.8.0`: https://github.com/tensorflow/tensorflow/blob/70b96696977c2f31a4b66aedc97353712430359e/ci/official/requirements_updater/requirements.in#L11C1-L11C8

Which would mean that merging this would mean you couldn't update to the newest pydantic if you use tensorflow.

I think we could justify this if it offered significant feature improvements (like was arguably the case when we bumped the minimum version to 4.6.1), but I think supporting deprecated fields doesn't quite rise to the level that justifies this level of friction.

I think a better route might be to keep the minimum version of `typing_extensions` as-is and more gracefully handle `typing_extensions.deprecated` in that case, I'll add some code change suggestions where appropriate.

Reply:
I think you are looking at their pinned dependencies, I've downloaded the latest tf wheel and it actually requires `typing-extensions>=4.6.6`. But your argument is still valid, so I'll apply the changes!

Reply:
Ah okay, I'm not surprised if I misinterpreted how they set version requirements for dependencies, I just know when we updated the min version of typing_extensions to 4.6.1 it was incompatible with tensorflow (dug through the issues reported when we did that and found that). So I was ready to assume that was still going to be an issue.

Anyway if you don't object, then yeah I guess it's easiest to just continue to support whatever versions are currently supported.